procfs/procfs_linux.go

```diff
@@ -0,0 +1,180 @@
+// +build linux
+
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package procfs
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+	"unicode"
+
+	"k8s.io/klog"
+)
+
+// ProcFS implements ProcInterface
+type ProcFS struct{}
+
+// NewProcFS creates ProcFS object
+func NewProcFS() ProcInterface {
+	return &ProcFS{}
```

Comment:
Don't return a pointer - that just wastes heap when it could stay stack-allocated